### PR TITLE
Object detection by distillation

### PR DESCRIPTION
**Pull request by faezehamjadi**

### CODE REVIEW

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+layout: review
+title:  "Knowledg Distillation on Object Detection"
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: Knowledge

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+cite:
+  authors: "Guobin Chen, Wongun Choi, Xiang Yu, Tony X. Han, Manmohan Krishna Chandraker"
+  title:   "Learning Efficient Object Detection Models with
+Knowledge Distillation"
+  venue:   "NIPS2017"
+---
+
+### Description
+	
+They proposed an end-to-end trainable framework for learning multi-class object detection through knowledge distillation. A teacher network (a pretrained network with more parameter and deeper) supervises and leads to better student performance which the knowledg is received from teacher is more informative.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: parameters*
knowledge*
which the knowledge is received from teacher is more informative.  (phrasing? It's not clear)

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	
+They proposed an end-to-end trainable framework for learning multi-class object detection through knowledge distillation. A teacher network (a pretrained network with more parameter and deeper) supervises and leads to better student performance which the knowledg is received from teacher is more informative. 
+
+<img src="/deep-learning/images/detecting_KD/0.png" width="800">
+
+
+Faster-RCNN is their object detection framework which is composed of two modules:1) a region proposal network(RPN) that generates object proposals, and 2) RPN inputs to a classification and regression network(RCN) that return the detection score and bounding boxes.They learn a student object detector by using Knowledge of a high capacity teacher detection network.There is three step:
+
+1) Encourage the feature representation of a student network is similar to that of the teacher network.
+2) The sof teacher targets in classification and regrestion is applied to transfer knowledge to student.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: sof? regrestion

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+3) The student have a more loss by groundtruth and its prediction.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: phrasing

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Faster-RCNN is their object detection framework which is composed of two modules:1) a region proposal network(RPN) that generates object proposals, and 2) RPN inputs to a classification and regression network(RCN) that return the detection score and bounding boxes.They learn a student object detector by using Knowledge of a high capacity teacher detection network.There is three step:
+
+1) Encourage the feature representation of a student network is similar to that of the teacher network.
+2) The sof teacher targets in classification and regrestion is applied to transfer knowledge to student.
+3) The student have a more loss by groundtruth and its prediction.   
+
+
+### Losses:
+
+The soft labels contain information about the relationship between different classes as discovered by teacher. By learning from soft labels, the student network inherits such hidden information.In object detection, the possible errors are misclassifications betweek background and foreground, they address this with a class-weighted cross entropy.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: between*

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+
+
+For classification, they apllied cross entropy losses in hard target(student prediction and groundtruth) and a class-weighted cross entropy for soft target( prediction between student and teacher)
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: applied*

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+<img src="/deep-learning/images/detecting_KD/1.png" width="250">
+
+<img src="/deep-learning/images/detecting_KD/2.png" width="350">
+
+For regression,because the real value regression outputs are unbounded, the teacher may provided a direction that is contrary to groundtruth.The authors, instead of using teacher's regression output as a target, they inforce The student’s regression to be as close to the ground truth label, but once the quality of the student surpasses that of the teacher with a certain margin, they
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: For regression, beca...

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+do not provide additional loss for the studen.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: student*

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+<img src="/deep-learning/images/detecting_KD/3.png" width="500">
+
+For similarity between features in intermidiate layer in student(V) and teacher(Z) network, they apply L2 distance:
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: intermediate

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+Finally the loss is :
+
+<img src="/deep-learning/images/detecting_KD/6.png" width="400">
+
+where N,M are the batch-size for RCN and RPN, respectively and $\lambda$ and $\gamma$ are hyper-parameter.
+
+
+### Experiments
+
+They applied two different setting. First, applying different networks size in student and teacher and second,with keeping the same model for both, smaller input image size for the student model and larger input image size for the teacher is applied.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: settings*

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+For similarity between features in intermidiate layer in student(V) and teacher(Z) network, they apply L2 distance:
+
+<img src="/deep-learning/images/detecting_KD/4.png" width="200">
+
+Finally the loss is :
+
+<img src="/deep-learning/images/detecting_KD/6.png" width="400">
+
+where N,M are the batch-size for RCN and RPN, respectively and $\lambda$ and $\gamma$ are hyper-parameter.
```
Comment (Dref360) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: Don't you need $$ instead of $ to do latex with Math.Js? (Not sure)

Comment (faezehamjadi) [reply deep-learning/_posts/2018-03-22-detectionDistillation.md]: you are right. thank you

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+### Description
+	
+They proposed an end-to-end trainable framework for learning multi-class object detection through knowledge distillation. A teacher network (a pretrained network with more parameters and deeper) supervises and leads to better student performance which the knowledge is received from teacher is more informative. 
+
+<img src="/deep-learning/images/detecting_KD/0.png" width="800">
+
+
+Faster-RCNN is their object detection framework which is composed of two modules:1) a region proposal network(RPN) that generates object proposals, and 2) RPN inputs to a classification and regression network(RCN) that return the detection score and bounding boxes.They learn a student object detector by using Knowledge of a high capacity teacher detection network.There is three steps:
+
+1) Encourage the feature representation of a student network is similar to that of the teacher network.
```
Comment (jodoin) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: 1.
2.
3.

replace ) by .

Comment (jodoin) [reply deep-learning/_posts/2018-03-22-detectionDistillation.md]: to be similar

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+2) The soft teacher targets in classification and regrestion is applied to transfer knowledge to student.
+3) The student have a more loss by groundtruth and its prediction.
```
Comment (jodoin) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: The student network has loss computed between its prediction and the groundtruth.

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+Faster-RCNN is their object detection framework which is composed of two modules:1) a region proposal network(RPN) that generates object proposals, and 2) RPN inputs to a classification and regression network(RCN) that return the detection score and bounding boxes.They learn a student object detector by using Knowledge of a high capacity teacher detection network.There is three steps:
+
+1) Encourage the feature representation of a student network is similar to that of the teacher network.
+2) The soft teacher targets in classification and regrestion is applied to transfer knowledge to student.
+3) The student have a more loss by groundtruth and its prediction.   
+
+
+### Losses:
+
+The soft labels contain information about the relationship between different classes as discovered by teacher. By learning from soft labels, the student network inherits such hidden information. In object detection, the possible errors are misclassifications between background and foreground, they address this with a class-weighted cross entropy.
```
Comment (jodoin) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: by the teacher.

Comment (jodoin) [reply deep-learning/_posts/2018-03-22-detectionDistillation.md]: foreground.  They address this issue...

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+
+
+For classification, they aplied cross entropy losses in hard target(student prediction and groundtruth) and a class-weighted cross entropy for soft target( prediction between student and teacher)
```
Comment (jodoin) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: applied a  cross entropy loss on hard...

File: deep-learning/_posts/2018-03-22-detectionDistillation.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+<img src="/deep-learning/images/detecting_KD/1.png" width="250">
+
+<img src="/deep-learning/images/detecting_KD/2.png" width="350">
+
+For regression, because the real value regression outputs are unbounded, the teacher may provided a direction that is contrary to groundtruth.The authors, instead of using teacher's regression output as a target, they inforce The student’s regression to be as close to the ground truth label, but once the quality of the student surpasses that of the teacher with a certain margin, they
```
Comment (jodoin) [inline deep-learning/_posts/2018-03-22-detectionDistillation.md]: provide

**Pull request merged**